src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs

```diff
@@ -101,6 +104,18 @@ public IEnumerable<IVsPackageMetadata> GetInstalledPackages()
                 });
         }
 
+        private async Task<FallbackPackagePathResolver> GetPackagesPathResolverAsync(BuildIntegratedNuGetProject project)
+        {
+            // To get packagesPath for build integrated projects, first read the packageSpec to know if
+            // RestorePackagesPath property was specified. If yes, then use that property to get packages path
+            // otherwise use global user cache folder from _settings.
+            var context = new DependencyGraphCacheContext();
+            var packageSpec = (await project.GetPackageSpecsAsync(context)).First();
+            var packagesPath = VSRestoreSettingsUtilities.GetPackagesPath(_settings, packageSpec);
+
+            return new FallbackPackagePathResolver(packagesPath, SettingsUtility.GetFallbackPackageFolders(_settings));
```

Comment:
This needs to be VSRestoreSettingsUtilities.GetFallbackFolders

SettingsUtility will have the same issue for fallback folders as it did for the global packages folder.